src/index.spec.ts

```diff
@@ -35,6 +44,25 @@ describe('ts-node', function () {
   it('should export the correct version', function () {
     expect(VERSION).to.equal(require('../package.json').version)
   })
+  it('should export all CJS entrypoints', function () {
+    // Ensure our package.json "exports" declaration allows `require()`ing all our entrypoints
+    // https://github.com/TypeStrong/ts-node/pull/1026
+    testsDirRequire.resolve('ts-node')
+    testsDirRequire.resolve('ts-node/package')
+    testsDirRequire.resolve('ts-node/dist/index')
+    testsDirRequire.resolve('ts-node/dist/bin')
+    testsDirRequire.resolve('ts-node/dist/bin-transpile')
+    testsDirRequire.resolve('ts-node/dist/bin-script')
+    testsDirRequire.resolve('ts-node/dist/bin-script-deprecated')
+    testsDirRequire.resolve('ts-node/dist/esm')
+    testsDirRequire.resolve('ts-node/register')
+    testsDirRequire.resolve('ts-node/register/index')
+    testsDirRequire.resolve('ts-node/register/files')
+    testsDirRequire.resolve('ts-node/register/transpile-only')
+    testsDirRequire.resolve('ts-node/register/type-check')
+    testsDirRequire.resolve('ts-node/tsconfig.schema.json')
+    testsDirRequire.resolve('ts-node/tsconfig.schemastore-schema.json')
```

Comment:
Cannot load them, because they do things like install require hooks.  But resolving should be enough to know that package.json "exports" is still compatible with legacy require() calls.

Reply:
Should we remove some of these in the next major version?

Reply:
Good point, I've already merged other breaking changes to master, so our next release will be major already. 

Do you want me to tweak the `"exports"` object to only include a few?

Off the top of my head, here's a whitelist of the ones we should keep, what do you think @blakeembrey ?

```
ts-node

// only reliably way to ask node for the root path of a dependency is Path.resolve(require.resolve('ts-node/package'), '..')
ts-node/package

// All bin entrypoints for people who need to augment our CLI: `node -r otherstuff ./node_modules/ts-node/dist/bin`
ts-node/dist/bin
ts-node/dist/bin-transpile
ts-node/dist/bin-script

// Must be require()able, obviously
ts-node/register
ts-node/register/files
ts-node/register/transpile-only
ts-node/register/type-check

// Maybe this is useful?  I dunno; I'd hate to break someone
ts-node/tsconfig.schema
```

Reply:
SGTM. I think we could safely omit `tsconfig.schema` but add it back if requested, but happy to keep it there for anyone using it as a manually referenced schema.

Reply:
I'm assuming the reason we ship with `tsconfig.schema` at all is because of `$ref` or `$schema` in `tsconfig.json`? Maybe we could just omit it and have people reference a static version hosted outside the package going forward?

Reply:
It's so you can add `"$schema": "./node_modules/ts-node/tsconfig.schema.json"` to your tsconfig file and *know* you are using the correct schema for the version of ts-node you are using.  For example, if you're using an old or experimental build that changes which ts-node  flags you are allowed to embed in a tsconfig, you'll want to refer to the bundled schema.  Even then, you're probably referencing it via filename, not `require()`.

For everyone else, their editor is using the schemastore schema by default, which already includes our stuff.

I'll omit it.